docker/packager/cctools/Dockerfile

```diff
@@ -58,8 +58,67 @@ RUN wget https://sourceware.org/pub/gdb/releases/gdb-$GDB_VERSION.tar.gz \
     && make install \
     && rm -fr gdb-$GDB_VERSION gdb-$GDB_VERSION.tar.gz
 
+#
+# OpenSSL FIPS
```

Comment:
The Dockertools file of "cctools" (which we use for cross-compiling) doesn't look like the right place for OpenSSL. Could we make a separate Dockerfile?

Reply:
Related to that:

I tried to setup a run a simple FIPS-enabled build by myself. I
- build a stand-alone OpenSSL,
- copied fips.so into the system provider dir
- and changed fipsmodule.cnf and openssl.cnf (both at the system level)
Unfortunately, I keep getting errors

```
libc++abi: terminating due to uncaught exception of type std::runtime_error: Failed to load 'fips' provider.
Aborted (core dumped)
```

It would be cool to include a brief step-by-step guide for local debugging which doesn't change the system env. It could be based on setting `OPENSSL_CONF` which we evaluate in OpenSSLInitializer.cpp.

Reply:
>copied fips.so into the system provider dir

Where exactly?

Reply:
/usr/lib/x86_64-linux-gnu/ossl-modules (based on l. 65 and the other docker files, e.g. docker/test/integration/base/Dockerfile)

Reply:
😄

Could we nevertheless please include a small example for testing without touching system dirs (as much as possible)?